File: BaekjoonAlgorithm/202012/BA_1697/BJAlgo_1697.py
from sys import stdin
from collections import deque

MAX = 100001
n, k = map(int, stdin.readline().split())
array = [0] * MAX  # 해당 위치(인덱스)를 방문하는데 걸린 거리


def bfs(v):
    q = deque([v])
    while q:
        now_pos = q.popleft()
        if now_pos == k:
            return array[now_pos]
        for next_pos in (now_pos - 1, now_pos + 1, now_pos * 2):
            if 0 <= next_pos < MAX and not array[next_pos]:  # 다음 위치가 범위 내에 있고 지금까지 방문한 적이 없는 경우
                array[next_pos] = array[now_pos] + 1
                q.append(next_pos)


print(bfs(n))

File: BaekjoonAlgorithm/202012/BA_1697/BJAlgo_1697.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Arrays;
import java.util.Deque;

public class BJAlgo_1697 {
    static int MAX = 100001;

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int[] nk = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        int[] array = new int[MAX];

        System.out.println(bfs(nk, array));
    }

    private static int bfs(int[] nk, int[] arr) {
        Deque<Integer> deque = new ArrayDeque<>();
        deque.add(nk[0]);

        while (!deque.isEmpty()) {
            int nowPos = deque.pollFirst();
            if (nowPos == nk[1]) {
                return arr[nowPos];
            }

            int[] tmp = {nowPos - 1, nowPos + 1, nowPos * 2};
            for (int nextPos : tmp) {
                if ((nextPos >= 0 && nextPos < MAX) && arr[nextPos] == 0) {
                    arr[nextPos] = arr[nowPos] + 1;
                    deque.add(nextPos);
                }
            }
        }

        return 0;
    }
}
